### PR TITLE
39 refactor code

### PR DESCRIPTION
**Pull request by ElhamMoharrami**

### CODE REVIEW

File: src/assets/apis/config.js [hunk 1]
```diff
@@ -1,3 +1,8 @@
-export const MOVIESURL=`http://localhost:8080/api/movies`;
-export const ARTISTSURL=`http://localhost:8080/api/crews`;
-export const  pageRangeDisplayed=3
\ No newline at end of file
+export const BASEURL = `http://localhost:8080/api`;
+export const pageRangeDisplayed = 3;
```
Comment (ghamilouie) [inline src/assets/apis/config.js:2]: It is better to move these constant values where you are using them

File: src/components/Listing/Listing.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import { fetchData } from "../../store/api-call";
+import ShowList from "../ShowList/ShowList";
+
+import classes from "./Listing.module.css";
+import { options } from "../../assets/apis/config";
+
+import { pageRangeDisplayed } from "../../assets/apis/config";
+import { RiseLoader } from "react-spinners";
+
+const Listing = (props) => {
```
Comment (ghamilouie) [inline src/components/Listing/Listing.js:16]: It is better to not use ing form for file name, isn't `PagedList` a better name?

File: src/components/UI/CustomLink.js [hunk 3]
```diff
@@ -6,16 +6,16 @@ import { useDispatch } from "react-redux";
 const CustomLink = ({ to, children, ...props }) => {
   const dispatch = useDispatch();
 
-  const onClickHandler = (event) => {
-    event.preventDefault();
-    dispatch(dataActions.clearData());
-    window.location.reload(false);
-  };
+  // const onClickHandler = (event) => {
```
Comment (ghamilouie) [inline src/components/UI/CustomLink.js:9]: Remove comments

File: src/store/index.js [hunk 4]
```diff
@@ -1,10 +1,9 @@
 import { configureStore } from "@reduxjs/toolkit";
```
Comment (ghamilouie) [inline src/store/index.js:1]: Isn't better to change ` index.js` name to somethings else ex. `store.js`

File: src/store/data-slice.js [hunk 5]
```diff
@@ -1,102 +1,37 @@
 import { createSlice } from "@reduxjs/toolkit";
+import { RingLoader } from "react-spinners";
 
 const initialState = {
-  data: [],
+  data: { content: [], pageCount: 0 },
```
Comment (ghamilouie) [inline src/store/data-slice.js:5]: why you don't set page default size? don't you get NaN in first call? Check network

File: src/store/api-call.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
+import { BASEURL } from "../assets/apis/config";
+
+const getDataRequest = async (url) => {
+  const response = await fetch(url);
+  const data = await response.json();
```
Comment (ghamilouie) [inline src/store/api-call.js:5]: Why you create local variable for data? you make it inline like response

File: src/pages/Movies.js [hunk 7]
```diff
@@ -1,8 +1,22 @@
...
+
+import "../assets/commonStyle.css";
+import MovieCard from "../components/MovieCard/MovieCard";
+import { movieActions } from "../store/data-slice";
 
 const Movies = () => {
-  return <MovieList />;
+  const data = useSelector((state) => state.movies.data);
+  const card = (item) => {
+    return <MovieCard movie={item} key={item.id} />;
```
Comment (ghamilouie) [inline src/pages/Movies.js:13]: Isn't it better move `key={item.id}` into ShowList, where iteration is?

File: src/pages/Artists.js [hunk 8]
```diff
@@ -1,8 +1,22 @@
...
+import ArtistCard from "../components/ArtistCard/ArtistCard";
+
+import { artistActions } from "../store/data-slice";
 
 const Artists = () => {
-  return <ArtistList />;
+  const data = useSelector((state) => state.crews.data);
+
+  const card = (item) => {
+    return <ArtistCard artist={item} key={item.id} />;
```
Comment (ghamilouie) [inline src/pages/Artists.js:13]: Isn't it better move key={item.id} into ShowList, where iteration is?

File: src/pages/MovieDetailPage/MovieDetailPage.js [hunk 9]
```diff
@@ -74,12 +81,11 @@ const MovieDetail = () => {
...
-        </div>
+
+        <ShowList data={artists} card={card} />
       </div>
     </>
   );
 };
 
-export default MovieDetail;
+export default MovieDetailPage;
```
Comment (ghamilouie) [inline src/pages/MovieDetailPage/MovieDetailPage.js:91]: Isn't it better to use simple short name? ex `MovieDetail` or `Movie`

File: src/pages/ArtistDetailPage/ArtistDetailPage.js [hunk 10]
```diff
@@ -52,12 +58,10 @@ const ArtistDetail = () => {
...
-          <ShowMovies movies={movies} />
-        </div>
+        <ShowList data={movies} card={card} />
       </div>
     </>
   );
 };
 
-export default ArtistDetail;
+export default ArtistDetailPage;
```
Comment (ghamilouie) [inline src/pages/ArtistDetailPage/ArtistDetailPage.js:67]: Isn't it better to use simple short name? ex `ArtistDetail` or `Artist`

File: src/components/Listing/Listing.js [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+      top: document.body.scrollHeight,
+      behavior: "smooth",
+    });
+  };
+
+  const dropdownCloseHandler = () => {
+    window.scrollTo(0, 0);
+  };
+
+  console.log(isLoading);
```
Comment (ghamilouie) [inline src/components/Listing/Listing.js:59]: It's better to remove debug logs before commit

File: src/pages/MovieDetailPage/MovieDetailPage.js [hunk 12]
```diff
@@ -1,24 +1,31 @@
...
+import classes from "./MovieDetailPage.module.css";
+import ShowList from "../../components/ShowList/ShowList";
+
+import { movieActions } from "../../store/data-slice";
 
-import classes from "./MovieDetail.module.css";
-import ShowArtists from "../ShowArtists/ShowArtists";
+const MovieDetailPage = () => {
+  const card = (item) => {
+    return <ArtistCard artist={item} key={item.id} />;
```
Comment (ghamilouie) [inline src/pages/MovieDetailPage/MovieDetailPage.js:17]: same point for `key={item.id}`

File: src/pages/MovieDetailPage/MovieDetailPage.js [hunk 13]
```diff
@@ -74,12 +81,11 @@ const MovieDetail = () => {
       </div>
       <div className={classes.crew}>
         <p className={classes["crew-title"]}>Movie Crew</p>
-        <div className={classes["poster-grid"]}>
-          <ShowArtists artists={artists} />
-        </div>
+
```
Comment (ghamilouie) [inline src/pages/MovieDetailPage/MovieDetailPage.js:84]: Remove extra line

File: src/pages/Movies.js (hunk 7 above)
Comment (ghamilouie) [inline src/pages/Movies.js:13]: same point for key={item.id}

File: src/store/api-call.js [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+    }
+  };
+};
+
+export const fetchDetail = (id, type, action) => {
+  return async (dispatch) => {
+    try {
+      const url = `${BASEURL}/${type}/${id}`;
+      const getDetail = await getDataRequest(url);
+      dispatch(
```
Comment (ghamilouie) [inline src/store/api-call.js:32]: Reformat your code and make one line

File: src/store/index.js [hunk 15]
```diff
@@ -1,10 +1,9 @@
...
 
-import comboSlice from "./combo-slice";
-import dataSlice from "./data-slice";
+import dataSlice, { artistsSlice, movieSlice } from "./data-slice";
```
Comment (ghamilouie) [inline src/store/index.js:3]: Remove extra import `dataSlice`

**Review by ghamilouie — CHANGES_REQUESTED** [review]
Some common Rules:

- Remove not used imports
- Remove console.logs
- Remove comments
- Remove extra lines
- Use better clear short names

File: src/assets/apis/config.js (hunk 1 above)
Comment (ElhamMoharrami) [reply src/assets/apis/config.js:2]: done

File: src/components/Listing/Listing.js (hunk 2 above)
Comment (ElhamMoharrami) [reply src/components/Listing/Listing.js]: done

File: src/components/UI/CustomLink.js (hunk 3 above)
Comment (ElhamMoharrami) [reply src/components/UI/CustomLink.js:9]: done

File: src/store/index.js (hunk 4 above)
Comment (ElhamMoharrami) [reply src/store/index.js:1]: done

Comment (ElhamMoharrami) [reply src/store/index.js:1]: done

File: src/store/data-slice.js (hunk 5 above)
Comment (ElhamMoharrami) [reply src/store/data-slice.js:5]: I checked, I don't get NAN

File: src/store/api-call.js (hunk 6 above)
Comment (ElhamMoharrami) [reply src/store/api-call.js:5]: I could return response.json I just made data to look cleaner. plus it is a common convention to store it in a local variable when not using .then

File: src/pages/MovieDetailPage/MovieDetailPage.js (hunk 12 above)
Comment (ElhamMoharrami) [reply src/pages/MovieDetailPage/MovieDetailPage.js]: done

File: src/pages/MovieDetailPage/MovieDetailPage.js (hunk 13 above)
Comment (ElhamMoharrami) [reply src/pages/MovieDetailPage/MovieDetailPage.js]: done

File: src/pages/Movies.js (hunk 7 above)
Comment (ElhamMoharrami) [reply src/pages/Movies.js:13]: done

File: src/store/api-call.js (hunk 14 above)
Comment (ElhamMoharrami) [reply src/store/api-call.js:32]: done

File: src/store/index.js (hunk 15 above)
Comment (ElhamMoharrami) [reply src/store/index.js:3]: done

**Pull request merged**